node/src/components/block_proposer.rs

```diff
@@ -412,22 +426,22 @@ impl BlockProposerReady {
             .ancestor_values()
             .iter()
             .flat_map(|block_payload| block_payload.deploys_and_transfers_iter())
+            .map(DeployOrTransferHash::into)
```

Comment:
(Unrelated to this PR, but I wonder whether we should `take_while(|hash| !self.contains_finalized(hash))` here: As soon as we hit a deploy we already _know_ is finalized, all older deploys must also already be known, and it would usually spare us the work of collecting basically all deploys of this era so far?)

Reply:
Makes sense to me :+1:

Reply:
Thanks! Sorry for the long delay! I was OOO and forgot I had clicked "request changes". :grimacing:

Reply:
But `past_deploys` are used later in a couple of places:
* `self.deps_resolved(deploy, past_deploys)
so if we end collecting early then we might decide that some of the deploys cannot be proposed b/c their dependencies haven't been resolved.

Reply:
But `deps_resolved` also calls `self.contains_finalized(dep)`, in addition to checking `past_deploys`, so it should be fine?

Reply:
Ah, you're right. It's safe then. Sorry.